pkg/build/builder/docker.go

```diff
@@ -147,47 +147,16 @@ func (d *DockerBuilder) checkSourceURI(testConnection bool) error {
 // is included in the build revision, that commit ID is checked out. Otherwise
 // if a ref is included in the source definition, that ref is checked out.
 func (d *DockerBuilder) fetchSource(dir string) error {
-	origProxy := make(map[string]string)
-	var setHttp, setHttps bool
-	// set the http proxy to be used by the git clone performed by S2I
-	if len(d.build.Spec.Source.Git.HTTPSProxy) != 0 {
-		glog.V(2).Infof("Setting https proxy variables for Git to %s", d.build.Spec.Source.Git.HTTPSProxy)
-		origProxy["HTTPS_PROXY"] = os.Getenv("HTTPS_PROXY")
-		origProxy["https_proxy"] = os.Getenv("https_proxy")
-		os.Setenv("HTTPS_PROXY", d.build.Spec.Source.Git.HTTPSProxy)
-		os.Setenv("https_proxy", d.build.Spec.Source.Git.HTTPSProxy)
-		setHttps = true
-	}
-	if len(d.build.Spec.Source.Git.HTTPProxy) != 0 {
-		glog.V(2).Infof("Setting http proxy variables for Git to %s", d.build.Spec.Source.Git.HTTPProxy)
-		origProxy["HTTP_PROXY"] = os.Getenv("HTTP_PROXY")
-		origProxy["http_proxy"] = os.Getenv("http_proxy")
-		os.Setenv("HTTP_PROXY", d.build.Spec.Source.Git.HTTPProxy)
-		os.Setenv("http_proxy", d.build.Spec.Source.Git.HTTPProxy)
-		setHttp = true
-	}
-
-	// can't test access to the git uri if a proxy is required.
-	if err := d.checkSourceURI(!setHttp && !setHttps); err != nil {
+	// Set the HTTP and HTTPS proxies to be used by git clone.
+	originalProxies := setHTTPProxy(d.build.Spec.Source.Git.HTTPProxy, d.build.Spec.Source.Git.HTTPSProxy)
+	defer resetHTTPProxy(originalProxies)
+
+	// Check source URI, trying to connect to the server only if not using a proxy.
+	usingProxy := len(originalProxies) > 0
+	if err := d.checkSourceURI(!usingProxy); err != nil {
```

Comment:
@bparees could you remind me why we can't use the proxy to test the connnectivity here?